commons/src/test/java/info/metadude/android/eventfahrplan/commons/SystemClockTest.kt

```diff
@@ -0,0 +1,108 @@
+package info.metadude.android.eventfahrplan.commons
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.threeten.bp.LocalDate
+import org.threeten.bp.ZoneOffset
+import org.threeten.bp.ZonedDateTime
+
+class SystemClockTest {
+
+    val Dec30_22_47_2019 = 1577746077615 // 2019-12-30T22:47:57.615Z
+
+	@Test
+    fun dateTimeFieldsAreCorrectlyMapped() {
+        val clock = SystemClock(Dec30_22_47_2019)
+
+        assertThat(clock.year).isEqualTo(2019)
+        assertThat(clock.month).isEqualTo(12)
+        assertThat(clock.monthDay).isEqualTo(30)
+        assertThat(clock.hour).isEqualTo(22)
+        assertThat(clock.minute).isEqualTo(47)
+    }
+
+    @Test
+    fun startOfDay() {
+        val clock = SystemClock(Dec30_22_47_2019)
+        val startOfDay = clock.startOfDay()
+
+        assertThat(startOfDay.year).isEqualTo(2019)
+        assertThat(startOfDay.month).isEqualTo(12)
+        assertThat(startOfDay.monthDay).isEqualTo(30)
+        assertThat(startOfDay.hour).isEqualTo(0)
+        assertThat(startOfDay.minute).isEqualTo(0)
+    }
+
+    @Test
+    fun endOfDay() {
+        val clock = SystemClock(Dec30_22_47_2019)
+        val endOfDayUTC = clock.endOfDay().toUTCDateTime()
+
+        assertThat(endOfDayUTC.year).isEqualTo(2019)
+        assertThat(endOfDayUTC.monthValue).isEqualTo(12)
+        assertThat(endOfDayUTC.dayOfMonth).isEqualTo(30)
+        assertThat(endOfDayUTC.hour).isEqualTo(23)
+        assertThat(endOfDayUTC.minute).isEqualTo(59)
+    }
+
+    @Test
+    fun getMinuteOfDayZonedInput() {
+        val startsAtDate = ZonedDateTime.of(2019, 8, 27, 6, 30, 0, 0, ZoneOffset.ofHours(4))
+        assertThat(SystemClock(startsAtDate).minuteOfDay).isEqualTo((6 - 4) * 60 + 30)
+    }
+
+    @Test
+    fun getDayOfMonthWithLeapYearDay() {
+        // Thursday, February 28, 2019 11:59:59 PM UTC
+        assertThat(SystemClock(1551312000000).monthDay).isEqualTo(28)
+    }
+
+    @Test
+    fun getDayOfMonthWithDayAfterLeapYear() {
+        // Friday, March 1, 2019 12:00:00 AM UTC
+        assertThat(SystemClock(1551398400000).monthDay).isEqualTo(1)
+    }
+
+    @Test
+    fun startOfDayVSLocalDate() {
```

Comment:
We should maybe define a project guideline (and fix all issues at once), if you think acronym casing is important. Because at the web I have not found any common sense / standard on this.

Reply:
Agreed, that would be fair.